### PR TITLE
Fix to use url-safe base64 in note url

### PR DESCRIPTION
**Pull request by jackycute**
fix #742

### CODE REVIEW

**Comment by jackycute** [comment]
@DenisValeev Please check the latest commit in this PR.

**Comment by DenisValeev** [comment]
Perfect

File: public/js/history.js [hunk 1]
```diff
@@ -291,6 +297,15 @@ function parseToHistory (list, notehistory, callback) {
...
   else if (notehistory && notehistory.length > 0) {
     for (let i = 0; i < notehistory.length; i++) {
+      // migrate LZString encoded id to base64url encoded id
+      try {
+        let id = LZString.decompressFromBase64(notehistory[i].id)
+        if (id && checkNoteIdValid(id)) {
+          notehistory[i].id = encodeNoteId(id)
+        }
+      } catch (err) {
+        // na
```
Comment (SISheogorath) [inline public/js/history.js]: Mhm, maybe at least some logging would be nice.

**Comment by jackycute** [comment]
5e975cbe690048e144fb6bd99c5b239a3e764445 also fix #727

**Comment by SISheogorath** [comment]
@jackycute do you think it's safe to merge? Or does it need further testing?

**Comment by jackycute** [comment]
It's good for now 👍

**Comment by SISheogorath** [comment]
Thanks a lot for your work @jackycute!

**Pull request merged**